solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -909,8 +884,8 @@ public void close() {
             });
 
     ExecutorUtil.shutdownAndAwaitTermination(notifications);
-    ExecutorUtil.shutdownAndAwaitTermination(collectionPropsNotifications);
-    if (closeClient) {
+    collectionPropertiesZkStateReader.close();
+    if (closeClient && zkClient != null) {
```

Comment:
why the zkClient null check?  It wasn't there before.

Reply:
Removed the check